### PR TITLE
Use xz for y2logs compression except 12-sp1

### PR DESCRIPTION
**Pull request by dzedro**
- Fail: https://openqa.suse.de/tests/3549026#step/install_update/43
- Verification run:
http://10.100.12.155/tests/13697
https://openqa.suse.de/tests/3552625 aarch64
https://openqa.suse.de/tests/3552634 ppc64le
https://openqa.suse.de/tests/3552632 x86_64
https://openqa.opensuse.org/tests/1075734#step/firefox/137 TW

### CODE REVIEW

**Comment by dzedro** [comment]
I would prefer just xz, but xz is not supported on 12sp1

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -417,8 +417,9 @@ sub export_logs {
     $self->save_and_upload_log('systemctl status',          '/tmp/systemctl_status.log');
     $self->save_and_upload_log('systemctl',                 '/tmp/systemctl.log', {screenshot => 1});
 
-    script_run "save_y2logs /tmp/y2logs_clone.tar.bz2";
-    upload_logs "/tmp/y2logs_clone.tar.bz2";
+    my $compression = is_sle('=12-sp1') ? 'bz2' : 'xz';
```
Comment (rwx788) [inline lib/opensusebasetest.pm]: We could reuse `get_available_compression` method from:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/y2_logs_helper.pm#L117

Comment (dzedro) [reply lib/opensusebasetest.pm]: xz is default compression, I would prefer to use xz.

Comment (rwx788) [reply lib/opensusebasetest.pm]: alright

**Comment by dzedro** [comment]
Why merge I just changed it :cry: 

```
diff --git a/lib/opensusebasetest.pm b/lib/opensusebasetest.pm
index fe1d6ba5a..02dc694f5 100644
--- a/lib/opensusebasetest.pm
+++ b/lib/opensusebasetest.pm
@@ -12,6 +12,7 @@ use version_utils qw(is_sle is_leap is_upgrade is_aarch64_uefi_boot_hdd is_tumbl
 use main_common 'opensuse_welcome_applicable';
 use isotovideo;
 use IO::Socket::INET;
+use y2_logs_helper 'get_available_compression';
 
 # Base class for all openSUSE tests
 
@@ -417,9 +418,8 @@ sub export_logs {
     $self->save_and_upload_log('systemctl status',          '/tmp/systemctl_status.log');
     $self->save_and_upload_log('systemctl',                 '/tmp/systemctl.log', {screenshot => 1});
 
-    my $compression = is_sle('=12-sp1') ? 'bz2' : 'xz';
-    script_run "save_y2logs /tmp/y2logs_clone.tar.$compression";
-    upload_logs "/tmp/y2logs_clone.tar.$compression";
+    script_run "save_y2logs /tmp/y2logs_clone.tar" . get_available_compression;
+    upload_logs "/tmp/y2logs_clone.tar" . get_available_compression;
     $self->investigate_yast2_failure();
 }
 
diff --git a/lib/qam.pm b/lib/qam.pm
index affe594d8..01d158c0c 100644
--- a/lib/qam.pm
+++ b/lib/qam.pm
@@ -18,7 +18,7 @@ use Exporter;
 use testapi;
 use utils;
 use List::Util 'max';
-use version_utils 'is_sle';
+use y2_logs_helper 'get_available_compression';
 
 our @EXPORT
   = qw(capture_state check_automounter is_patch_needed add_test_repositories ssh_add_test_repositories remove_test_repositories advance_installer_window get_patches check_patch_variables);
@@ -29,9 +29,8 @@ use constant ZYPPER_STATUS_COL  => 5;
 sub capture_state {
     my ($state, $y2logs) = @_;
     if ($y2logs) {    #save y2logs if needed
-        my $compression = is_sle('=12-sp1') ? 'bz2' : 'xz';
-        assert_script_run "save_y2logs /tmp/y2logs_$state.tar.$compression";
-        upload_logs "/tmp/y2logs_$state.tar.$compression";
+        assert_script_run "save_y2logs /tmp/y2logs_$state.tar" . get_available_compression;
+        upload_logs "/tmp/y2logs_$state.tar" . get_available_compression;
         save_screenshot();
     }
     #upload ip status
diff --git a/lib/y2_logs_helper.pm b/lib/y2_logs_helper.pm
index 68c290391..5cf0e3a5d 100644
--- a/lib/y2_logs_helper.pm
+++ b/lib/y2_logs_helper.pm
@@ -115,8 +115,8 @@ sub verify_license_translations {
 }
 
 sub get_available_compression {
-    my %extensions = (bzip2 => '.bz2', gzip => '.gz', xz => '.xz');
-    foreach my $binary (sort keys %extensions) {
+    my %extensions = (xz => '.xz', bzip2 => '.bz2', gzip => '.gz');
+    foreach my $binary (keys %extensions) {
         return $extensions{$binary} unless script_run("type $binary");
     }
     return "";
```

**Comment by rwx788** [comment]
Sad, let's create another PR =)

**Comment by rwx788** [comment]
But, FYI, in hash you cannot really predict the order of the keys, so having `xz` as default would not work.

**Pull request merged**